fastparquet/writer.py

```diff
@@ -1243,11 +1351,120 @@ def merge(file_list, verify_schema=True, open_with=default_open,
     -------
     ParquetFile instance corresponding to the merged data.
     """
-    out = api.ParquetFile(file_list, verify_schema, open_with, root)
+    out = ParquetFile(file_list, verify_schema, open_with, root)
     out._write_common_metadata(open_with)
     return out
 
 
+def update(dirpath, data, row_group_offsets=None, sort_pnames:bool=True,
+           compression=None, open_with=default_open, mkdirs=None, rename=None,
+           remove_with=None, stats=True):
+    """Merge new data to existing parquet dataset.
```

Comment:
So what exactly is the correspondence between this and the new methods on ParquetFile? Do we need both?

Reply:
`ParquetFile.write_row_groups()` 'only':
- adds new data to an existing dataset,
- define where to insert new row groups in row group list,
- can reorder partition files (rename them),
- modify metadata

`update` is a 'macro' function that rolls out currently only one logic for updating a partitioned dataset (I identify a different update logic in the case of an ordered dataset, having no partition). It is the re-implementation of `append=overwrite` feature.
- it identifies partition files to be removed based on partition values identified in new data,
- it calls `ParquetFile.remove_row_groups()` to remove them
- it defines the logic to sort row groups that will be added from new data
- it calls `ParquetFile.write_row_groups()` to add new data (and `ParquetFile.write_row_groups()` 

I would not say there is correspondence between both.
- `update` is the implementation of an update logic for partitioned datasets.
- `ParquetFile.write_row_groups` 'only' add new row groups to an existing dataset.